zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java

```diff
@@ -389,4 +390,32 @@ private static final ZanataIdentity getIdentity() {
         }
         return null;
     }
+
+    /*****************************************************************************************
+     * HTTP request rules
+     ******************************************************************************************/
+
+    /**
+     * Check if user can access to REST URL with httpMethod.
+     * If request is from anonymous user(account == null),
+     * only 'Read' action are allowed. Otherwise, role base check will should be
+     * perform in related RestService class.
```

Comment:
That should be "**Additionally**, role-based check will be performed in the REST service class".  At least that's the way it should be working.

First we should check whether you can talk to our API at all, then we should check whether you can perform the specific API action you requested.